azure-iot-hub-devicesdk/azure/iot/hub/devicesdk/transport/mqtt/mqtt_provider.py

```diff
@@ -157,6 +159,8 @@ def publish(self, topic, message_payload):
         """
         logger.info("sending")
         message_info = self._mqtt_client.publish(topic=topic, payload=message_payload, qos=1)
+        if message_info.rc != mqtt.MQTT_ERR_SUCCESS:
+            raise Exception(mqtt.error_string(message_info.rc))
```

Comment:
>Exception [](start = 18, length = 9)

Can we define a custom Exception for this? Otherwise, there's no good way to catch it.

Reply:
Good point.  I added a sticky to define a custom exception for this.